newsfragments/1825.bugfix.rst

```diff
@@ -0,0 +1 @@
+Allow passing dist-info-dir to the bdist_wheel command, as well as to the PEP-517 build backend. Metadata can therefore now be injected prior to wheel building when following PEP-517 -- by :user:`pelson`
```

Comment:
Thank you very much for the contribution @pelson.

How about something like the following:

```suggestion
Re-use pre-existing ``.dist-info`` dir when creating wheels via the build backend APIs (PEP 517) and the `metadata_directory` argument is passed -- by :user:`pelson`.
```

Maybe that reads better? Because we don't want people using the `bdist_wheel` command as a CLI tool, we should probably avoid announcing that you can pass extra options to it and keep that information as part of the now private API...